utils/MonitoredResource.md

```diff
@@ -0,0 +1,83 @@
+
+
+### GCP_GKE_CONTAINER
+
+If the environment variable `KUBERNETES_SERVICE_HOST` is set.
+
+| Property Name  | Environment Variable | Metadata Request                 |
+|----------------|----------------------|----------------------------------|
+| gcp_account    |                      | project/project-id               |
+| cluster_name   |                      | instance/attributes/cluster-name |
+| namespace_id   | NAMESPACE            |                                  |
+| instance_id    |                      | instance/id                      |
+| pod_id         | HOSTNAME             |                                  |
+| container_name | CONTAINER_NAME       |                                  |
+| zone           |                      | instance/zone                    |
+
+The namespace_id and container_name are optional. We cannot get their value from environment
+variables unless k8s users expose them via the [Downward API][DownwardAPI]. See k8s
+[documentation][K8SDocumentation] and [code sample][K8SCodeSample].
+
+### GCP_GCE_INSTANCE
+
+If the GCP metadata service returns a value for "instance/id" and not GCP_GKE_CONTAINER.
+
+| Property Name  | Environment Variable | Metadata Request   |
+|----------------|----------------------|--------------------|
+| gcp_account    |                      | project/project-id |
+| instance_id    |                      | instance/id        |
+| zone           |                      | instance/zone      |
+
+### AWS_EC2_INSTANCE
+ 
+If the AWS metadata service returns a value for "instance_id".
+
+| Property Name  | Environment Variable | Metadata Request           |
+|----------------|----------------------|----------------------------|
+| aws_account    |                      | instance-identity/document |
+| instance_id    |                      | instance_id                |
+| region         |                      | instance-identity/document |
+
+The value return by the `instance-identity/document` metadata request is a document described 
+[here][AWSMetadataIdentityDocument].
+
+[AWSMetadata]: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html#instancedata-data-retrieval
+[AWSMetadataIdentityDocument]: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-identity-documents.html
+[DownwardAPI]: https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/
+[GCPMetadata]: https://cloud.google.com/compute/docs/storing-retrieving-metadata
+[GCPMetadataJavaExmple]: https://github.com/GoogleCloudPlatform/google-cloud-java/blob/master/google-cloud-core/src/main/java/com/google/cloud/MetadataConfig.java
+[K8SDocumentation]: https://cloud.google.com/kubernetes-engine/docs/tutorials/custom-metrics-autoscaling#exporting_metrics_from_the_application
+[K8SCodeSample]: https://cloud.google.com/kubernetes-engine/docs/tutorials/custom-metrics-autoscaling#exporting_metrics_from_the_application
```

Comment:
Sorry I pasted the wrong link just now. The link to code sample should be https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/blob/master/custom-metrics-autoscaling/direct-to-sd/sd_dummy_exporter.go.

Reply:
Fixed.